inference-engine/src/gna_plugin/gna_graph_tools.hpp

```diff
@@ -220,9 +220,14 @@ inline std::pair<InferenceEngine::CNNLayerPtr, int>  CNNNetCheckNextLayerSkipCer
             separate_layers(getInputTo(layer->outData[i]));
         }
 
+        std::set< CNNLayerPtr > visited;
```

Comment:
If you change a type of container for currentSet, you will not have to deal with duplicates.

Reply:
I don't see it. IMO it is not the problem with duplicates, but circular dependency.
Please see that element is inserted to `currentSet` in `separate_layers` lambda.
So after removal of the element the same element can be inserted once again (we have to track which elements were visited).

Reply:
Ok, it makes sense.